src/ripple/app/misc/Validations.cpp

```diff
@@ -387,6 +387,35 @@ class ValidationsImp : public Validations
         return ret;
     }
 
+    std::uint32_t getValidationTime (uint256 const& hash, int minValidations)
+    {
+        if (minValidations < 1)
+            minValidations = 1;
+
+        std::vector <std::uint32_t> times;
+
+        {
+            ScopedLockType sl (mLock);
+
+            auto j = findSet (hash);
+            for (auto& it : *j)
+            {
+                if (it.second->isTrusted())
+                    times.push_back (it.second->getSignTime());
+            }
+        }
+
+        if (times.size() < minValidations)
+            return 0;
+
+        if (times.size() == 1)
+            return times[0];
```

Comment:
The code below works for `times.size() == 1`, and the general case code is unlikely to be a bottle neck. I'd remove the special case.